src/App.tsx

```diff
@@ -15,36 +17,83 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  String = 'alphabetical',
+  Number = 'length',
+  Default = '',
+}
+
 export const App: React.FC = () => {
+  const [sortOrder, setSortOrder] = useState<SortType>(SortType.Default);
+  const [reversed, setReversed] = useState<boolean>(false);
+  const sortedGoods = [...goodsFromServer].sort((good1: string, good2: string) => {
+    switch (sortOrder) {
+      case 'alphabetical':
+        return good1.localeCompare(good2);
+      case 'length':
+        return good1.length - good2.length;
+      default:
+        return 0;
+    }
+  });
+
+  if (reversed) {
+    sortedGoods.reverse();
+  }
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          type="button"
+          className={cn('button', 'is-info', {
+            'is-light': sortOrder !== 'alphabetical',
+          })}
+          onClick={() => setSortOrder(SortType.String)}
+        >
           Sort alphabetically
         </button>
 
-        <button type="button" className="button is-success is-light">
+        <button
+          type="button"
+          className={cn('button', 'is-success', {
+            'is-light': sortOrder !== 'length',
+          })}
+          onClick={() => setSortOrder(SortType.Number)}
+        >
           Sort by length
         </button>
 
-        <button type="button" className="button is-warning is-light">
+        <button
+          type="button"
+          className={cn('button', 'is-warning', {
+            'is-light': !reversed,
+          })}
+          onClick={() => setReversed(!reversed)}
+        >
           Reverse
         </button>
 
-        <button type="button" className="button is-danger is-light">
-          Reset
-        </button>
+        {(reversed || sortOrder) && (
+          <button
+            type="button"
+            className="button is-danger"
+            onClick={() => {
+              setSortOrder(SortType.Default);
+              setReversed(false);
+            }}
+          >
+            Reset
+          </button>
+        )}
       </div>
 
       <ul>
-        <ul>
-          <li data-cy="Good">Dumplings</li>
-          <li data-cy="Good">Carrot</li>
-          <li data-cy="Good">Eggs</li>
-          <li data-cy="Good">Ice cream</li>
-          <li data-cy="Good">Apple</li>
-          <li data-cy="Good">...</li>
-        </ul>
+        {sortedGoods.map((good: string) => (
+          <li data-cy="Good" key={good}>
```

Comment:
[CODE KNOWLEDGE] You are rendering a list and using the 'good' as a key. This might cause issues if the goods are not unique. Consider using a unique id or the index of the item as a key.